File: fixed_point.m
function fixed_point()
    x = [input('x0: ')];
    TC = input('TC: ');
    epsilon = [];
    i = 2;
    while true
        x(i) = f(x(i - 1));
        epsilon(i - 1) = abs((x(i) - x(i-1)) / x(i)) * 100;
        if epsilon(i - 1) < TC
            fprintf("approximate root: %f, iterations: %d", x(i), i - 1);
            break;
        end
        i = i + 1;
    end                            
end

function result = f(x)
    result = exp(-x);
end

File: fixed_point.py
from math import exp

def f(x):
	result = exp(-x)
	return result

def float_input(text):
	while True:
		try:
			value = float(input(text))
			return value
		except Exception as exception:
			print(exception)

def fixed_point():
	x = [float_input("x0: ")]
	TC = float_input("TC: ")
	if True:
		epsilon = []
		i = 1
		while True:
			x.append(f(x[i - 1]))
			epsilon.append(abs((x[i] - x[i - 1]) / x[i]) * 100)
			if epsilon[i - 1] < TC:
				return {"approximate root": x[i], "iterations": i}
			i += 1
		
print(fixed_point())
		